File: leetcode/2245-destroying-asteroids/destroying-asteroids.cpp
class Solution {
public:
    bool asteroidsDestroyed(int mass, vector<int>& asteroids) {
        int n = asteroids.size();
        sort(asteroids.begin(), asteroids.end());
        bool check = true;
        long long curr = mass;

        for (long long i = 0; i < n; i++) {
            if (curr >= asteroids[i]) {
                curr += asteroids[i];
            } else {
                check = false;
            }
        }

        return check;
    }
};

File: leetcode/2245-destroying-asteroids/destroying-asteroids.py
class Solution:
    def asteroidsDestroyed(self, mass: int, asteroids: List[int]) -> bool:
        new_ass = sorted(asteroids)
        print(new_ass)
        tracker = mass
        print(tracker)
        for i in range(len(new_ass)):
            print(tracker)
            if tracker < new_ass[i]:
                return False
            tracker += new_ass[i]

        return True

        